drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +95,67 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5.4 Vol 4 Part E
+ * @param packet_buff	Pointer to a HCI packet buffer
+ * @return Length of the HCI packet in bytes, zero if no valid packet found.
+ */
+static uint16_t packet_len(const uint8_t *packet_buff)
+{
+	uint16_t payload_len = 0;
+	uint8_t header_len = 0;
+	const uint8_t *packet_hdr = &packet_buff[sizeof(packet_type)];
+	const uint8_t packet_type = packet_buff[0];
+
+	switch (packet_type) {
+	case H4_CMD: {
+		const struct bt_hci_cmd_hdr *hdr = packet_hdr;
+
+		/* Parameter Total Length */
+		payload_len = hdr->param_len;
+		header_len = BT_HCI_CMD_HDR_SIZE;
+		break;
```

Comment:
Final comment: Just a thought, not a request:
We could also just do
```suggestion
		return sizeof(packet_type) + BT_HCI_CMD_HDR_SIZE + hdr->param_len;
```

and similar for all cases, to omit the 2 local variables (`payload_len` and `header_len`). This could reduce the size of this function by 14 lines 

Up to you if you want to change this, or leave it as is